diff --git a/ldo.c b/ldo.c
index c563b1d..db64220 100644
--- a/ldo.c
+++ b/ldo.c
@@ -55,9 +55,13 @@
 #if defined(__cplusplus) && !defined(LUA_USE_LONGJMP)	/* { */
 
 /* C++ exceptions */
+#include <exception>
+#include <cstdio>
 #define LUAI_THROW(L,c)		throw(c)
 #define LUAI_TRY(L,c,a) \
-	try { a } catch(...) { if ((c)->status == 0) (c)->status = -1; }
+	try { a } \
+	catch(std::exception &cranked_ex) { fprintf(stderr, "LUAI_TRY native exception: %s\n", cranked_ex.what()); fflush(stderr); if ((c)->status == 0) (c)->status = -1; } \
+	catch(...) { if ((c)->status == 0) (c)->status = -1; }
 #define luai_jmpbuf		int  /* dummy variable */
 
 #elif defined(LUA_USE_POSIX)				/* }{ */
